app/controllers/settings/external_cert_controller.rb

```diff
@@ -161,17 +187,20 @@ def cert_parse(cert_string)
     else
       cert = read_cert(cert_string)
       unless cert
-        # :nocov:
+        # :nocov: Temporary blocking nocov statement until tests are completed
         # Cert has aleady been valiated when entered, an error here would require a failure
         # in connection from velum to salt or an unintended change in the salt pillar
         params[:Message] = { Error: "Failed to parse stored certificate, please check format and " \
         "upload again" }
         return params
-        # :nocov:
+        # :nocov: Temporary blocking nocov statement until tests are completed
       end
       fingerprint = cert_fingerprint(cert)
+      san_hash = get_san_hash(cert)
+      altnames = san_hash[:altnames] || []
+      ip_altnames = san_hash[:ip_altnames] || []
```

Comment:
Could you extract all these parsing methods to a dedicated decorator class? Because we'll need to reuse these in the pages we are exposing the cert content itself and not just the info as you are doing here. This will also decrease the size/complexity of this controller file that is already 500+ lines.